matrix/rules.py

```diff
@@ -303,8 +305,10 @@ def exception_handler(self, context, loop=None, exc_ctx=None):
         reporter = functools.partial(self.exception_handler, context)
         self.loop.set_exception_handler(reporter)
         try:
+            await context.juju_model.connect_current()
```

Comment:
Agreed.  `libjuju` doesn't currently have that functionality, though.  I had a TODO comment in here but I accidentally dropped it during a refactor.